src/components/Timeline/__tests__/index.jsx

```diff
@@ -71,4 +91,83 @@ describe('<Timeline />', () => {
     wrapper.find(Header).prop('onLeave')()
     expect(wrapper.state('pointerHighlighted')).toBe(false)
   })
+
+  it('ensures the header scroll is in sync with the timeline when it is sticky', () => {
+    raf.mockImplementation(cb => cb())
+
+    const props = createProps({ isHeaderSticky: true })
+    const wrapper = mount(<Timeline {...props} />)
+    expect(wrapper.state('scrollLeft')).toBe(0)
+    wrapper.getDOMNode().scrollLeft = 50
+
+    wrapper.simulate('scroll')
+    expect(wrapper.state('scrollLeft')).toBe(50)
+  })
+
+  it('calls getMarkerOffset() and getTimelineWidth() when mounted if the timeline has a sticky header', () => {
+    const getMarkerOffset = jest.fn()
+    const getTimelineWidth = jest.fn()
+    const props = createProps({ getMarkerOffset, getTimelineWidth, stickyHeader: true })
+    const wrapper = mount(<Timeline {...props} />)
+    expect(getMarkerOffset).toBeCalledWith(wrapper.node.timeline)
+    expect(getTimelineWidth).toBeCalledWith(wrapper.node.timeline)
+  })
+
+  it('adds the resize event listener when mounted if the timeline has a sticky header', () => {
+    addListener.mockImplementation(jest.fn())
+
+    const props = createProps({ stickyHeader: true })
+    const wrapper = mount(<Timeline {...props} />)
+    expect(addListener).toBeCalledWith('resize', wrapper.instance().handleResize)
+  })
+
+  it('removes the resize event listener when unmounted if the timeline has a sticky header', () => {
+    removeListener.mockImplementation(jest.fn())
+
+    const props = createProps({ stickyHeader: true })
+    const wrapper = mount(<Timeline {...props} />)
+    wrapper.instance().componentWillUnmount()
+    expect(removeListener).toBeCalledWith('resize', wrapper.instance().handleResize)
+  })
+
+  it('does not attempt to remove the resize event listener if the timeline does not have a sticky header', () => {
+    removeListener.mockImplementation(jest.fn())
+
+    const props = createProps({ stickyHeader: false })
+    const wrapper = mount(<Timeline {...props} />)
+    wrapper.instance().componentWillUnmount()
+    expect(removeListener).not.toBeCalledWith('resize', wrapper.instance().handleResize)
+  })
+
+  it('gets the timeline width during resize events', () => {
+    addListener.mockImplementation((event, cb) => cb())
+    raf.mockImplementation(cb => cb())
```

Comment:
Could these implementations be done at the time of mocking, if they're not directly tested here?